src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidatorProvider.cs

```diff
@@ -16,11 +18,42 @@ namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
     /// </summary>
     public class DataAnnotationsModelValidatorProvider : IModelValidatorProvider
     {
+        private IOptions<MvcDataAnnotationsLocalizationOptions> _options;
+        private IStringLocalizerFactory _stringLocalizerFactory;
+
+        public DataAnnotationsModelValidatorProvider(
+            IOptions<MvcDataAnnotationsLocalizationOptions> options,
+            IStringLocalizerFactory stringLocalizerFactory)
+        {
+            _options = options;
+            _stringLocalizerFactory = stringLocalizerFactory;
+        }
+
         public void GetValidators(ModelValidatorProviderContext context)
         {
+            IStringLocalizer stringLocalizer = null;
+            if (_options != null &&
+                _options.Value.DataAnnotationLocalizerProvider != null &&
+                _stringLocalizerFactory != null)
+            {
+                stringLocalizer = _options.Value.DataAnnotationLocalizerProvider(
+                    context.ModelMetadata.ContainerType ?? context.ModelMetadata.ModelType,
```

Comment:
Primitive data types like `string`, `int` are also treated as `ModelType`. We don't want to create `.resx` for each data types/property. `ContainerType` is the Complex type. So we want to look at `ModelType` only if `ContainerType` is `null`